### PR TITLE
BaSP-TG-44: Timesheets Screens

### PR DESCRIPTION
**Pull request by JuliMonte5**
Screens for Timesheets resource, with the following functionalities:

- See the list of timesheets saved in the DB.
- Edit a timesheet.
- Delete a timesheet.
- Add a new timesheet.

### CODE REVIEW

File: src/Components/TimeSheets/Expand Modal/expandmodal.module.css [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    width: 100%;
+    background-color: rgba(0,0,0,0.5);
+}
+
+.modal {
+    display: flex;
+    flex-direction: column;
+    width: 500px;
+    height: fit-content;
+    background-color: white;
```
Comment (ignacioaristo) [inline src/Components/TimeSheets/Expand Modal/expandmodal.module.css:20]: lets use hex convention.

File: src/Components/TimeSheets/Form/index.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    }
+  };
+
+  const timeSheetFetch = async (id) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/timeSheet/${id}`);
+      const data = await response.json();
+      setTimeSheet(data.data);
+    } catch (error) {
+      console.error(error);
```
Comment (ignacioaristo) [inline src/Components/TimeSheets/Form/index.js:67]: Same with this console.error

**Review by ignacioaristo — CHANGES_REQUESTED** [review]
1. Change al the component to a arrow functions (ES6).
2. The folder's name must be in camelCase.
3. We are not handle the success and error messages. For example when we delete a timesheet or create or update.

File: src/Components/TimeSheets/Form/index.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  const [Hours, setHours] = useState('');
+  const [ProjectId, setProjectId] = useState('');
+  const [EmployeeId, setEmployeeId] = useState('');
+  const [TaskId, setTaskId] = useState('');
+  const [timeSheet, setTimeSheet] = useState({});
+  const path = window.location.pathname.split('/');
+  let timeSheetId = path[path.length - 1];
+  let formSwitch;
+
+  timeSheetId.length == 24 ? (formSwitch = true) : (formSwitch = false);
```
Comment (Ignacio-Sormani) [inline src/Components/TimeSheets/Form/index.js:20]: Two comments related to this:
1) Don't use this kind of conditionals for functionality. It is normally use as conditions inside of return method. Use IFs
2) Don't add logic by itself in the component. It should be inside a function or useEffect.

File: src/Components/TimeSheets/Form/index.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+  const [Tasks, setTasks] = useState([]);
+  const [Description, setDescription] = useState('');
+  const [Date, setDate] = useState('');
+  const [Hours, setHours] = useState('');
+  const [ProjectId, setProjectId] = useState('');
+  const [EmployeeId, setEmployeeId] = useState('');
+  const [TaskId, setTaskId] = useState('');
+  const [timeSheet, setTimeSheet] = useState({});
+  const path = window.location.pathname.split('/');
+  let timeSheetId = path[path.length - 1];
```
[the remark this answers is not in the document]
Comment (Ignacio-Sormani) [reply src/Components/TimeSheets/Form/index.js:17]: If the value will change use it as a state. For example the formSwitch variable should be a state

**Review by Ignacio-Sormani — CHANGES_REQUESTED** [review]

File: src/Components/TimeSheets/Form/index.js [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+      hours: Hours
+    };
+    if (formSwitch) {
+      await fetch(`${process.env.REACT_APP_API_URL}/timeSheet/${timeSheetId}`, {
+        method: 'PUT',
+        headers: {
+          'Content-type': 'application/json'
+        },
+        body: JSON.stringify(req)
+      });
```
Comment (fede230298) [inline src/Components/TimeSheets/Form/index.js:97]: Manage the different status responses, both success and errors after performing the fetch

**Review by Ignacio-Sormani — APPROVED** [review]

File: src/Components/TimeSheets/index.js [hunk 6]
```diff
@@ -1,10 +1,123 @@
...
+    }
+  }, []);
+
+  const deleteTimesheet = async (id, timeSheets) => {
+    const response = await fetch(`${process.env.REACT_APP_API_URL}/timeSheet/${id}`, {
+      method: 'DELETE'
+    });
+    if (response.status === 204) {
+      setTimeSheets([...timeSheets.filter((timeSheetItem) => timeSheetItem._id !== id)]);
+      alert('Used deleted');
```
Comment (MaicolFreire001) [inline src/Components/TimeSheets/index.js:29]: This alert message seems wrong

**Review by MaicolFreire001 — CHANGES_REQUESTED** [review]

**Review by MaicolFreire001 — APPROVED** [review]
Seems good to me

**Review by ignacioaristo — APPROVED** [review]

**Review by javierperezraffo — APPROVED** [review]

**Pull request merged**